src/JustEat.StatsD/StatsDPublisher.cs

```diff
@@ -34,7 +42,12 @@ public StatsDPublisher(StatsDConfiguration configuration, IStatsDTransport trans
             }
         }
 
-        public StatsDPublisher(StatsDConfiguration configuration, bool preferBufferedTransport = false)
+        public StatsDPublisher(StatsDConfiguration configuration)
+            : this(configuration, false)
+        {
+        }
+
+        public StatsDPublisher(StatsDConfiguration configuration, bool preferBufferedTransport)
```

Comment:
What's the reasoning behind making "better performance" opt-in? I think it would be better to add a constructor to accept the new interface, and have the existing two try use the better implementation, if possible, by default.

Reply:
As I have added new transport interface implementation to the existing transports, adding a constructor with signature accepting transport would result in a compilation error for the consumer.

I will make buffered implementation a default though.

Reply:
I'm not sure whether it's a good idea or not, but if you swapped the parameter order for the new one, it wouldn't be ambiguous to the compiler?

Reply:
The compiler will accept it without issues. 😸 
Looks hacky though. 

Other options I can think of: 
- Static factory methods on StatsDPublisher
- Dedicated factory class
- Dedicated transports per interface
- Expose new public implementation of BufferdPublisher, String publisher will retain the old name